src/gseproject/templates/game/shop.html

```diff
@@ -9,67 +9,116 @@
-
-          // Reset selected tools
-          for (var i = 0; i < selectedTools.length; i++) {
-              var element = document.getElementById(selectedTools[i].elementId);
-              element.style.backgroundColor = '#fae113'; // Reset background color
-          }
-          selectedTools = []; // Clear the array
-      } else {
-          alert("Not enough cherries!");
-      }
-  }
+    var selectedTools = []; // Array to store selected tools
+
+    function selectTool(cherryCount, element) {
+        // Change background color of the selected tool
+        element.style.backgroundColor = '#e18d00';
+        element.style.padding = '2px';
+        element.style.paddingBottom = '8px';
+        element.style.paddingTop = '8px';
+        element.style.borderRadius = '10px';
+
+        // Add the selected tool and its cherry count to the array
+        selectedTools.push({ cherryCount: cherryCount, elementId: element.id });
+    }
+
+    function deductCherryCount() {
+        var totalCherryCount = 0;
+
+        // Calculate the total cherry count
+        for (var i = 0; i < selectedTools.length; i++) {
+            totalCherryCount += selectedTools[i].cherryCount;
+        }
+
+        var cherryNumElement = document.getElementById("cherry_num");
+        var cherryNumValue = parseInt(cherryNumElement.textContent);
+
+        // Check if there are enough cherries
+        if (cherryNumValue >= totalCherryCount) {
+            // Deduct the cherry count
+            cherryNumValue -= totalCherryCount;
+            cherryNumElement.textContent = cherryNumValue;
+
+            // Reset selected tools
+            for (var i = 0; i < selectedTools.length; i++) {
+                var element = document.getElementById(selectedTools[i].elementId);
+                element.style.backgroundColor = '#fae113'; // Reset background color
+            }
+            selectedTools = []; // Clear the array
+        } else {
+            alert("Not enough cherries!");
+        }
+    }
+
+    function startPetWalk() {
+        fetch('{% url "start-pet-walk" %}', {
+            method: 'POST',
+            headers: {
+                'X-CSRFToken': getCookie('csrftoken'),
+                'Content-Type': 'application/json',
+            },
+            body: JSON.stringify({}),
+        })
+            .then((response) => {
+                if (response.ok) {
+                    redirectToPage('/game')
+                    petDisappear();
+                    return response.json();
+                }
+                throw new Error('Network response was not ok.');
+            })
+            .then((data) => {
+                console.log(data);
+                deductCherryCount();
+            })
+            .catch((error) => {
+                console.error('There has been a problem with your fetch operation:', error);
+            });
+    }
+    // Function to get CSRF token
+    function getCookie(name) {
+
```

Comment:
**This is reapeated code. Remove it.**

Reply:
This a better version of indentation, syntax changes only

Reply:
Then it should be changed in it's orginal location